From 348f7ed0cd355451408b5206f8fa423d406bc440 Mon Sep 17 00:00:00 2001
From: Frediano Ziglio <fziglio@redhat.com>
Date: Sat, 30 Jun 2018 05:13:07 +0100
Subject: [PATCH 43/43] Use GetModuleHandle to get some functions from user32
 library

The library is surely already loaded as providing clipboard and
other utilities we need.
user32 is one of the main win32 libraries.
Avoid using LoadLibrary that increment the reference so we don't
need to call FreeLibrary to cleanly decrement the reference.

Signed-off-by: Frediano Ziglio <fziglio@redhat.com>
Acked-by: Christophe Fergeau <cfergeau@redhat.com>
---
 vdagent/vdagent.cpp | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/vdagent/vdagent.cpp b/vdagent/vdagent.cpp
index be17ddc..89019bb 100644
--- a/vdagent/vdagent.cpp
+++ b/vdagent/vdagent.cpp
@@ -125,7 +125,6 @@ private:
     static VDAgent* _singleton;
     HWND _hwnd;
     HWND _hwnd_next_viewer;
-    HMODULE _user_lib;
     PCLIPBOARD_OP _add_clipboard_listener;
     PCLIPBOARD_OP _remove_clipboard_listener;
     clipboard_owner_t _clipboard_owner;
@@ -183,7 +182,6 @@ VDAgent* VDAgent::get()
 VDAgent::VDAgent()
     : _hwnd (NULL)
     , _hwnd_next_viewer (NULL)
-    , _user_lib (NULL)
     , _add_clipboard_listener (NULL)
     , _remove_clipboard_listener (NULL)
     , _clipboard_owner (owner_none)
@@ -222,7 +220,6 @@ VDAgent::VDAgent()
 
 VDAgent::~VDAgent()
 {
-    FreeLibrary(_user_lib);
     close_vio_serial();
     CloseHandle(_stop_event);
     CloseHandle(_control_event);
@@ -285,9 +282,9 @@ bool VDAgent::run()
         vd_printf("SetProcessShutdownParameters failed %lu", GetLastError());
     }
 
-    _user_lib = LoadLibrary(L"User32.dll");
+    HMODULE _user_lib = GetModuleHandle(L"User32");
     if (!_user_lib) {
-        vd_printf("LoadLibrary failed %lu", GetLastError());
+        vd_printf("GetModuleHandle failed %lu", GetLastError());
         return false;
     }
     _add_clipboard_listener =
-- 
2.17.1

